mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/DefaultEntityFactory.java

```diff
@@ -347,4 +352,80 @@ private CqlNameGenerator buildCqlNameGenerator(TypeElement classElement) {
     }
     return new TypeMirror[0];
   }
+
+  private boolean isTransient(
+      String propertyName,
+      Set<String> transientProperties,
+      TypeElement classElement,
+      ExecutableElement getMethod,
+      VariableElement field,
+      int partitionKeyIndex,
+      int clusteringColumnIndex) {
+
+    // check if property is present in @TransientProperties
+    Class<?> otherAnnotationClass =
+        partitionKeyIndex >= 0
+            ? PartitionKey.class
+            : clusteringColumnIndex >= 0 ? ClusteringColumn.class : null;
```

Comment:
We should also check other mapper annotations: `CqlName`, `Computed` and maybe others in the future. I'm not sure if passing a value for each annotation will scale well, so maybe we could just re-read them on the field and getter.
```java
for (Class<?> annotationClass : ImmutableList.of(PartitionKey.class, ClusteringColumn.class, CqlName.class) {
  if (getMethod.getAnnotation(annotationClass) != null || field != null && field.getAnnotation(annotationClass) != null) {
    ...
  }
}
```
We could extract that into another method so that it's easier to call multiple times in this method (see my other comment).
```java
void checkTransientNotAnnotated(
      ExecutableElement getMethod,
      VariableElement field,
      Element elementToWarnOn)
```
Also possibly report all offending annotations if there is more than one.

Reply:
I like this idea, I think we could take it one step further and use it for more than just transient as I will also face the same issue with `@Computed`.  I realized that the 3.x mapper must have similar logic to prevent multiple conflicting annotations from being used on a property and that is the case: 

https://github.com/datastax/java-driver/blob/3.x/driver-mapping/src/main/java/com/datastax/driver/mapping/AnnotationChecks.java#L53-L93

I think the solution there works but is a bit more complicated than it needs to be.  I"ll try to come up with something.

Reply:
Actually, there are parts of it that I like a lot, such as resolving an annotation in a general way (i.e. pass both the field and the method to resolve the annotation), which will be nice for custom inheritence strategies when we add it.   I'll take a stab at refactoring things in a way to meet my needs, without trying to complicate things too much.